docs/tutorials/golang/getting-started.md

```diff
@@ -0,0 +1,183 @@
+
+	return ctx, err
+}
+```
+
+## Hub user authorization
+
+Next, you can provide access to your Hub APIs to your users by generating an API token for them. Tokens are based on your user's identity public keys and your `context` created above. You could use any keypair identity, but we'll use the simple `threads.Identity` type shown above.
+
+```go
+package main
+
+import (
+	"context"
+	"github.com/textileio/go-threads/api/client"
+	"github.com/textileio/go-threads/core/thread"
+	"google.golang.org/grpc"
+)
+
+var HUB_API = "api.textile.io:3447"
+var HUB_HEADERS = []grpc.DialOption{grpc.WithInsecure(), grpc.WithPerRPCCredentials(common.Credentials{})}
+
+func authorizeUser(ctx context.Context, user thread.Identity) (context.Context, error) {
+	// Create an API Client
+	cli, err := client.NewClient(HUB_API, HUB_HEADERS...)
+	if err != nil {
+		return nil, err
+	}
+  // Generate a new token for the user
+  tok, err := cli.GetToken(ctx, user)
+	if err != nil {
+		return nil, err
+	}
+  // // Add the token to the context
+	ctx = thread.NewTokenContext(ctx, tok)
+	return ctx, nil
+}
+```
+
+## Create threads with the client
+
+The client allows you to create fully remote threads on the Hub. This is a good approach for cases where the application doesn't require fully offline modes.
+
+```go
+package main
+
+import (
+	"context"
+  "time"
+	crand "crypto/rand"
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/textileio/go-threads/core/thread"
+	"github.com/textileio/go-threads/common"
+	"google.golang.org/grpc"
+)
+
+var HUB_API = "api.textile.io:3447"
+var HUB_HEADERS = []grpc.DialOption{grpc.WithInsecure(), grpc.WithPerRPCCredentials(common.Credentials{})}
+
+func createThreadDB(ctx context.Context, name string) error {
+  // Assign the name for the Thread
+  ctx = common.NewThreadNameContext(ctx, name)
+  // Generate our client again
+	cli, err := client.NewClient(HUB_API, HUB_HEADERS...)
+	if err != nil {
+		return err
+  }
+  // Create the ID
+	dbID := thread.NewIDV1(thread.Raw, 32)
+  // Create the database
+	return cli.NewDB(ctx, dbID)
+}
+```
+
+## Create a Bucket
+
+Creating buckets in Go applications shares many of the same steps as creating threads. Because buckets are managed in a thread, you must first create a thread where you will then add a bucket. A single thread can store all of your user's buckets (and more).
+
+```go
+package main
+
+import (
+	"context"
+  "time"
+	crand "crypto/rand"
+	"github.com/textileio/go-threads/core/thread"
+	"github.com/textileio/go-threads/common"
+	buckets "github.com/textileio/textile/api/buckets/client"
+	"google.golang.org/grpc"
+)
+
+var HUB_API = "api.textile.io:3447"
+var HUB_HEADERS = []grpc.DialOption{grpc.WithInsecure(), grpc.WithPerRPCCredentials(common.Credentials{})}
+
+func createBucket(ctx context.Context, name string) error {
+  // Create a new threaddb for our buckets
+  ctx = common.NewThreadNameContext(ctx, "my-buckets")
```

Comment:
Note that with the upcoming private buckets PR, some of this stuff will change slightly. But IIRC, there is also a change to how names are used? Not 100% of the consequences there yet though.